src/insights_client/metrics.py

```diff
@@ -90,13 +126,19 @@ def __init__(
             try:
                 auth_method = cfg.get("insights-client", "authmethod")
             except configparser.NoOptionError:
+                logger.debug("Metrics: authmethod not defined in insights-client config. Defaulting to CERT")
                 auth_method = AUTH_METHOD_CERT
 
             if auth_method == AUTH_METHOD_BASIC:
                 self.base_url = "cloud.redhat.com"
                 self.port = 443
-                u = cfg.get("insights-client", "username")
-                p = cfg.get("insights-client", "password")
+                try:
+                    u = cfg.get("insights-client", "username")
+                    p = cfg.get("insights-client", "password")
+                except configparser.NoOptionError:
+                    logger.debug("Metrics: BASIC auth selected but username and/or password is missing. No metrics will be sent.")
+                    self.offline = True
+                    return
```

Comment:
If we are adding this, I think the same should happen for an unknown auth_method.

```python
if auth_method == AUTH_METHOD_BASIC:
    ...
elif auth_method == AUTH_METHOD_CERT:
    ...
else:
    logger.debug("Metrics: Unknown authentication method. No metrics will be sent.")
    self.offline = True